connectivity/check/check.go

```diff
@@ -72,6 +72,9 @@ type Parameters struct {
 
 	CollectSysdumpOnFailure bool
 	SysdumpOptions          sysdump.Options
+
+	// PrometheusServeAddr is the exposed port on cilium agent to retrieve metrics.
+	PrometheusServeAddr string // TODO DONIA: Not sure where to store it
```

Comment:
I think should define an interface here, that tells us what pods to collect from (e.g. agent, operator, hubble etc) on which port.

It could be something like:

```suggestion
	CollectMetrics []MetricsSource
```

Where the interface is:

```go
type MetricsSource interface {
   func Pods() ([]*corev1.Pod, error)
   func Port() string
}
```

And then implement this interface e.g. for the agent metrics. The expectations also should specify where they want to collect metrics from and what metrics they actually care about.

```go
WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
	return check.ResultOK.ExpectMetricsIncrease(a.CiliumAgentMetrics(), "cilium_forward_count_total"), check.ResultOk
})
```

Where as `CiliumAgentMetrics` defines a `MetricsSource`. This is just a rough sketch and needs to be fleshed out (i.e. I don't expect my proposal to work as is), but generally speaking the idea here should be that expectations declare what metrics they are interested in, and that in turn tells the framework not only that it needs to collect metrics, but also where it needs to collect metrics from.

Happy to discuss this further offline.